### PR TITLE
Generate import libraries for free-threaded Python 3.13+ builds

### LINKED ISSUE #80: Generating import library for Python 3.13t free-threaded DLL
I think we might need a separate `python313t.def` as well as generation options for the free-threaded build.

My reason for thinking this: symbols like [`PyUnstable_Module_SetGIL`](https://github.com/python/cpython/blob/3afb639f39e89888194d8e74cc498c8da3a58d8e/Include/moduleobject.h#L104), which PyO3 does use, are only defined on the free-threaded build.

So I think we correspondingly might need to specifically generate a free-threaded import library? I'll try to find an example which doesn't work with the current setup.

### PR DESCRIPTION
**Pull request by ravenexp**
- Collect `python313t.dll` from Windows CI builds.
- Add new `ImportLibraryGenerator::abi_tag()` method
- Add unit tests and update documentation

Resolves #80

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -175,6 +185,11 @@ pub struct ImportLibraryGenerator {
     version: Option<(u8, u8)>,
     /// Python interpreter implementation
     implementation: PythonImplementation,
+    /// Optional Python ABI tag
+    ///
+    /// For example, `"t"` stands for the free-threaded CPython v3.13 build
+    /// aka CPython `3.13t`.
+    abi_tag: Option<String>,
```
Comment (davidhewitt) [inline src/lib.rs]: I think the more normal name I've seen for these might be "abiflags". That's at least what they're called in `sysconfig` and `maturin`. I think the full "abi tag" would be `cp313t` or `abi3` based upon [the packaging spec](https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/#platform-compatibility-tags).

```suggestion
    abiflags: Option<String>,
```

File: src/lib.rs [hunk 2]
```diff
@@ -202,6 +218,17 @@ impl ImportLibraryGenerator {
         self
     }
 
+    /// Sets the ABI tag for the `pythonXY<abi>.dll` import library.
+    ///
+    /// For example, `"t"` stands for the free-threaded CPython v3.13 build
+    /// aka CPython `3.13t`. `python313t.dll` import library is generated.
+    ///
+    /// The untagged versioned `pythonXY.dll` is generated by default.
+    pub fn abi_tag(&mut self, abi_tag: Option<&str>) -> &mut Self {
```
Comment (davidhewitt) [inline src/lib.rs]: Similar suggestion:

```suggestion
    pub fn abiflags(&mut self, abi_tag: Option<&str>) -> &mut Self {
```

**Review by davidhewitt — COMMENTED** [review]
Thanks very much for the rapid turnaround! I think string flags is probably fine imo, both `sysconfig` and `maturin` just handle these as strings.

I was able to put together https://github.com/PyO3/pyo3/pull/4749 to test this out downstream; I was able to reproduce a broken build and that this patch fixes it! 👍

File: src/lib.rs (hunk 1 above)
Comment (ravenexp) [reply src/lib.rs]: Applied.

**Review by messense — APPROVED** [review]
Thanks!

File: src/lib.rs (hunk 2 above)
Comment (ravenexp) [reply src/lib.rs]: Applied.

**Comment by ravenexp** [comment]
> Thanks very much for the rapid turnaround! I think string flags is probably fine imo, both `sysconfig` and `maturin` just handle these as strings.
> 
> I was able to put together [PyO3/pyo3#4749](https://github.com/PyO3/pyo3/pull/4749) to test this out downstream; I was able to reproduce a broken build and that this patch fixes it! 👍

I've renamed the builder method and am going to merge this and release a new crate version after adding some more tests for Windows ARM64.

**Review by davidhewitt — APPROVED** [review]
Super, thanks 👍

**Pull request merged**